tests/random_test.py

```diff
@@ -1643,30 +1643,29 @@ class KeyArrayTest(jtu.JaxTestCase):
   # might also be a more general test of opaque element types. If
   # so, add a corresponding test to to CustomElementTypesTest as well.
 
-  def make_keys(self, *shape, seed=None):
-    if seed is None:
-      seed = 28
+  def make_keys(self, *shape, seed=28):
     seeds = seed + jnp.arange(math.prod(shape), dtype=jnp.uint32)
-    make_key = partial(prng.seed_with_impl, prng.threefry_prng_impl)
-    return jnp.reshape(jax.vmap(make_key)(seeds), shape)
+    return random.key(seeds).reshape(shape)
```

Comment:
Can we actually make `random.key` scalar-valued only (like `PRNGKey`) for now, and not rely on its batch-polymorphism yet? I'm noticing now that I didn't include a scalar check in `random.key`, but we could add it still.

An outstanding consideration is that we may want to accept more-than-one-int as the seed to `key`, since the key is typically more than one int of information. This might warrant accepting an array to that end instead of for batching.

Reply:
Done, thanks! I'll also send a separate PR to make `random.key` error on non-scalar inputs.

Reply:
Addressed in #16484